mobile/src/main/java/ch/epfl/sdp/mobile/state/game/delegating/DelegatingTextToSpeechState.kt

```diff
@@ -0,0 +1,67 @@
+package ch.epfl.sdp.mobile.state.game.delegating
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import ch.epfl.sdp.mobile.application.chess.engine.Board
+import ch.epfl.sdp.mobile.application.chess.engine.Color
+import ch.epfl.sdp.mobile.application.chess.engine.Piece
+import ch.epfl.sdp.mobile.application.chess.engine.rules.Action
+import ch.epfl.sdp.mobile.application.tts.TextToSpeechFacade
+import ch.epfl.sdp.mobile.infrastructure.tts.TextToSpeech
+import ch.epfl.sdp.mobile.state.game.core.GameDelegate
+import ch.epfl.sdp.mobile.ui.game.TextToSpeechState
+import kotlin.properties.Delegates.notNull
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class DelegatingTextToSpeechState
+constructor(
+    private val gameDelegate: GameDelegate,
+    private val facade: TextToSpeechFacade,
+    private val scope: CoroutineScope,
+) : TextToSpeechState {
+
+  override var muted by mutableStateOf(false)
```

Comment:
Consider storing the state of whether TTS is muted as a preference in the facade.